src/main/kotlin/reminder/domain/ReminderRepository.kt

```diff
@@ -1,5 +1,8 @@
 package reminder.domain
 
-fun interface ReminderRepository {
+import java.util.*
+
+interface ReminderRepository {
     fun save(reminder: Reminder)
+    fun search(identifier: ReminderIdentifier): Optional<Reminder>
```

Comment:
I don't know if it is better to have a separate interface for each use case (because mabye someone just wants to implement the save functionality?).
Anyway, it is a decision that we can do in the future, as for know having more methods on the interface doesn't bother us because we use mockito for mocking the interfaces (and we don't have to implement the methods by hand on our own mocks).